File: Bases para conhecimento do AG JSSP/AGclassico/helloWorldAG/Individuo.java
package helloWorldAG;

import java.util.Random;

public class Individuo {

    private String genes = "";
    private int aptidao = 0;

    //gera um indivíduo aleatório
    public Individuo(int numGenes) {
        genes = "";
        Random r = new Random();
        
        String caracteres = Algoritimo.getCaracteres();
        
        for (int i = 0; i < numGenes; i++) {
            genes += caracteres.charAt(r.nextInt(caracteres.length()));
        }
        
        geraAptidao();        
    }

    //cria um indivíduo com os genes definidos
    public Individuo(String genes) {    
        this.genes = genes;
        
        Random r = new Random();
        //se for mutar, cria um gene aleatório
        if (r.nextDouble() <= Algoritimo.getTaxaDeMutacao()) {
            String caracteres = Algoritimo.getCaracteres();
            String geneNovo="";
            int posAleatoria = r.nextInt(genes.length());
            for (int i = 0; i < genes.length(); i++) {
                if (i==posAleatoria){
                    geneNovo += caracteres.charAt(r.nextInt(caracteres.length()));
                }else{
                    geneNovo += genes.charAt(i);
                }
                
            }
            this.genes = geneNovo;   
        }
        geraAptidao();
    }

    //gera o valor de aptidão, será calculada pelo número de bits do gene iguais ao da solução
    private void geraAptidao() {
        String solucao = Algoritimo.getSolucao();
        for (int i = 0; i < solucao.length(); i++) {
            if (solucao.charAt(i) == genes.charAt(i)) {
                aptidao++;
            }
        }
    }

    public int getAptidao() {
        return aptidao;
    }

    public String getGenes() {
        return genes;
    }
}

File: Bases para conhecimento do AG JSSP/AGclassico/Individuo.cs
using System;

namespace helloWorldAG
{

	public class Individuo
	{

		private string genes = "";
		private int aptidao = 0;

		//gera um indivíduo aleatório
		public Individuo(int numGenes)
		{
			genes = "";
			Random r = new Random();

			string caracteres = Algoritimo.Caracteres;

			for (int i = 0; i < numGenes; i++)
			{
				genes += caracteres[r.Next(caracteres.Length)];
			}

			geraAptidao();
		}

		//cria um indivíduo com os genes definidos
		public Individuo(string genes)
		{
			this.genes = genes;

			Random r = new Random();
			//se for mutar, cria um gene aleatório
			if (r.NextDouble() <= Algoritimo.TaxaDeMutacao)
			{
				string caracteres = Algoritimo.Caracteres;
				string geneNovo = "";
				int posAleatoria = r.Next(genes.Length);
				for (int i = 0; i < genes.Length; i++)
				{
					if (i == posAleatoria)
					{
						geneNovo += caracteres[r.Next(caracteres.Length)];
					}
					else
					{
						geneNovo += genes[i];
					}

				}
				this.genes = geneNovo;
			}
			geraAptidao();
		}

		//gera o valor de aptidão, será calculada pelo número de bits do gene iguais ao da solução
		private void geraAptidao()
		{
			string solucao = Algoritimo.Solucao;
			for (int i = 0; i < solucao.Length; i++)
			{
				if (solucao[i] == genes[i])
				{
					aptidao++;
				}
			}
		}

		public virtual int Aptidao
		{
			get
			{
				return aptidao;
			}
		}

		public virtual string Genes
		{
			get
			{
				return genes;
			}
		}
	}
}